frontend/src/components/AuthManager.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, FunctionComponent } from 'react';
+import { GoogleLogin } from 'react-google-login';
+import { useHistory, useLocation } from "react-router-dom";
+import useClientId from '../hooks/useClientId';
+import AuthContext from '../context/auth';
+import LandingPage from './LandingPage';
+
+export const AuthManager: FunctionComponent = ({ children }) => {
+  const [signedIn, setSignedIn] = useState(false);
+  const clientId = useClientId();
+  let history = useHistory();
+  const { pathname } = useLocation();
+
+  function logout() {
+    localStorage.clear();
+    setSignedIn(false);
+    history.push('/');
+  }
+
+  async function onSignIn(googleUser: any) {
+    let token = googleUser.getAuthResponse().id_token;
+    let email = googleUser.profileObj.email;
+
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body:
+        JSON.stringify({
+          "token": token,
+          "email": email,
+          "table": "Dispatchers",
+          "clientId": clientId
+        })
+    };
+
+    const authorized = await fetch('/auth', requestOptions)
+      .then(res => res.json())
+      .then(data => data['id']);
+
+    if (authorized) {
+      setSignedIn(true);
+      if (pathname === '/') {
+        history.push('/dashboard/home');
+      }
+    } else {
+      logout();
+    }
+  }
+
+  return (
```

Comment:
i would personally use inline ifs here, but is there an advantage to using a ternary statement?

Reply:
Cleaned it up a bit, but just thought that the ternary really enforces that this is an if-else situation. Probably would have used && in situations where the ternary would be `variable ? something : <></>` to eliminate the unnecessary fragment.

Reply:
yeah i got what youre saying, makes sense to me 👍